src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -2193,26 +2342,44 @@ ErrorIfUnsupportedAggregateDistinct(Aggref *aggregateExpression,
 		}
 	}
 
+	if (aggregateType == AGGREGATE_COUNT)
```

Comment:
Oops, the change in this function seems to allow other aggregates in the subquery, such as `SUM`. And, the result of the execution seems wrong compared to the local table:

```SQL


SELECT 
    l_partkey, sum_of_order_keys  
FROM 
  (SELECT 
        sum(distinct l_orderkey) AS sum_of_order_keys, l_partkey
  FROM 
      lineitem
  GROUP BY
     l_partkey) as repartitioned_table 
ORDER BY 
  sum_of_order_keys, l_partkey 
DESC LIMIT 10;

 l_partkey | sum_of_order_keys 
-----------+-------------------
    195853 |             17994
    149605 |             17994
    101036 |             17994
     57771 |             17994
     19353 |             17994
    166895 |             17996
    119794 |             17996
     58188 |             17996
      3110 |             17996
    150058 |             17998
(10 rows)

SELECT 
    l_partkey, sum_of_order_keys  
FROM 
  (SELECT 
        sum(distinct l_orderkey) AS sum_of_order_keys, l_partkey
  FROM 
      lineitem_local 
  GROUP BY
     l_partkey) as repartitioned_table 
ORDER BY 
  sum_of_order_keys, l_partkey 
DESC LIMIT 10;

 l_partkey | sum_of_order_keys 
-----------+-------------------
    195853 |              8997
    149605 |              8997
    101036 |              8997
     57771 |              8997
     19353 |              8997
    166895 |              8998
    119794 |              8998
     58188 |              8998
      3110 |              8998
    150058 |              8999
(10 rows)
```